src/src/com/microsoft/aad/adal/BrokerProxy.java

```diff
@@ -93,7 +94,7 @@ public BrokerProxy(final Context ctx) {
      * does not direct call if the caller is from Authenticator itself.
      */
     @Override
-    public boolean canSwitchToBroker() {
+    public boolean canSwitchToBroker() throws DeveloperAuthenticationException {
```

Comment:
I hope we can move the verifyManifestPermission to the end of the check. If the signature of the broker even doesn't match, then does it make sense to throw an error and tell the calling app that some permissions are missing?